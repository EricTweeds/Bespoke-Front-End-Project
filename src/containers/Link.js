import React, { Component }  from 'react'
import { NavLink } from 'react-router-dom'

const style = {
    textDecoration: 'none'
}

export default class Weather extends Component {
	render() {
        let link = '/' + this.props.filter
        return (
            <NavLink to={link} style={style}>
                { this.props.children }
            </NavLink>
        )
    }
}