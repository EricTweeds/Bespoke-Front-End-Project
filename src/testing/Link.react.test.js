import React from 'react'
import Link from './Link.react'
import renderer from 'react-test-renderer'

test('Link changes the class when hovered', () => {
    const component = renderer.create(
        <Link page="https://github.com/EricTweeds/Bespoke-Front-End-Project">Repo</Link>
    )
    //test default
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    //test with mouse hover
    tree.props.onMouseEnter()
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    //test after mouse leaves
    tree.props.onMouseLeave()
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})