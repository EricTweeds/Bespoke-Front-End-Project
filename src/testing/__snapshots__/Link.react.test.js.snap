// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="https://github.com/EricTweeds/Bespoke-Front-End-Project"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Repo
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="https://github.com/EricTweeds/Bespoke-Front-End-Project"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Repo
</a>
`;

exports[`Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="https://github.com/EricTweeds/Bespoke-Front-End-Project"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Repo
</a>
`;
